### DISCUSSION TITLE
2022-03-02 Meeting Agenda, Recording & Notes

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3650] [DDEV Advisory Group]
# Agenda

* Turn on recording, ask for a minutes-taker
* Set next meeting date (or recurring)
* Announcements about DDEV trademark, sponsorship (both GitHub and other), etc.
* Followup from budget group meeting [https://docs.google.com/document/d/1YxU6N5m2qVfVJSSYXgjBbdPM6nBz9X4YjR0DWZIeMLw/edit?usp=sharing](https://www.google.com/url?q=https://docs.google.com/document/d/1YxU6N5m2qVfVJSSYXgjBbdPM6nBz9X4YjR0DWZIeMLw/edit?usp%3Dsharing&sa=D&source=calendar&ust=1646669014553696&usg=AOvVaw2VpYk7I4kPETfNJtVCUsTE)
* Review of v1.19 features (coming soon)
* Questions:
  * When should PHP8.0 become the default PHP version?
  * Should any other defaults change in upcoming ddev v1.19.0?
  * Highest priorities for v1.20?
  * Guidance on test-runner replacement (Windows/macOS)
  * Nominations for upstream projects to donate to with 10% of GitHub Sponsors

# Minutes

Based on the Agenda from the Meeting Invite, we will add notes under each top-level bullet:

- Turn on recording, ask for a minutes-taker
  - chris wells will spearhead this
- Set next meeting date (or recurring)
  - May 4th at 11A EST.
- Announcements about DDEV trademark, sponsorship (both GitHub and other), etc.
  - Open a GitHub Sponsors on the rfay personal - and the response has been amazing, and sits at about 33% of $5000/mo. We can solve a number of problems with that already. And believe it or not, that&#39;s the small news. :)
  - About a week after that set of events, platform.sh approached Randy and said they would like to be the &quot;lead sponsor&quot; of the project (or at least A lead sponsor). We talked about a number of ways to do that. The preference was to pay Randy a salary to be the lead maintainer of ddev. That&#39;s a huge thing. In the process, they have also agreed with Fruition (deed not quite signed yet, perhaps), but they&#39;ve agreed to purchase the Trademark and other assets from Fruition. The uncertainty about the DDEV name and all those things are just at the edge of being solved. Those two things are not a fully done deal. We have a couple problems to solve with employments contracts, etc, but those are being worked on in good faith. Will be sorted out in the coming days.
  - Introduction from Fabien - talk about this from the perspective of Platform.sh. Fabien Potencier, creator of Symfony framework. He is Chief Product Officer of Platform.sh - joined when Blackfire was acquired by Platform.
    - Why Platform help DDEV? We provide a hosting platform and we provide a CLI tool. The CLI tool is written in PHP right now, and for many years we&#39;ve wanted to write it in Go. We now have more Python customers, etc - so they don&#39;t want a PHP tool. A Go-based tool is now a requirement for us. Also want to provide a full dev stack with Docker, and here we are. Looked at several open source projects, and DDEV was the one we liked the most. A few weeks ago talked with Brad &amp; Randy and we want to help the project &amp; community.
    - Two deals -
      - 1, secure the brand, trademark, domain name, etc from Fruition. We finalized that deal a few minutes ago.
      - 2, also agreed to sponsor Randy for his work on DDEV.
    - We care about long-term version for DDEV that Randy has. We also have many ideas but we want to be responsible for the project also. I think we know how it works. Want to keep flexibility with DDEV, do NOT want to remove Acquia/Pantheon integration. But, we do want good/better integration with platform.sh. Want to sponsor more generic support for Python and NodeJS, which might also be useful for PHP users also, or users outside Platform.
  - Randy - We want to communicate this news to the larger community in the best way.
    - Cary - this is great, thank you. You&#39;re not &quot;buying DDEV&quot; but you are buying Randy&#39;s time to continue to work on it. The concern is that ownership by one of the big three hosts might be scary in terms of &quot;this one&#39;s ours now.&quot;
    - FP - correct, we&#39;re not &quot;buying&quot; anything (just the trademark). Or not an acquisition. Words are important, we&#39;re here as a sponsor.
    - Cary - what does &quot;buying the trademark mean?&quot;
    - Brad - Fruition stepped in to make sure there wasn&#39;t uncertainty around the project. We approached Platform because it does fit in their business model. As a community member, I do not expect Platform to make this &quot;the Platform product.&quot; Fruition will still continue to use DDEV and their hosting platform, wouldn&#39;t have worked with Platform unless they were sure it was securing the longevity of the project.
  - Cary - we still want to come up with governance for the open source project.
  - Truls - what I&#39;ve seen so far here is great. But I have some practical questions-like… when? Will the main maintainer be employed? Will you be sponsoring the Github? How will all that work?
    - Randy - the way Platform sponsored me by employing me with a salary to work on DDEV.
    - The GitHub org is going to have Fabien as a rep from Platform and Randy as the official owners of the Org. Platform has formally agreed to act in Good Faith if anything ever happens to them. Ex.g. employment agreement doesn&#39;t work, then Platform has said &quot;well Randy stays as role of owner in org&quot; etc. If something happens to the company, or they cannot sponsor any more - they will act in Good Faith with the trademark, etc.
  - Truls - thanks. Do we have two options? That&#39;s a good option, but a Git plan so you can sponsor directly to GitHub in different levels.
    - One thing we did before this came about, we added GitHub sponsors to Randy personally and we have 33% of $5000/mo. The money there goes into LocalDev Foundation.
    - Randy&#39;s contract with Platform is exclusive, and he cannot take money from the LocalDev foundation. So now, that money may be able to go to someone else, OR be used for infrastructure purchases, etc. GitHub sponsorship will move to the (new?) org for DDEV.
  - Cary - in fact, yes, there can be some serious repercussions about all this money now. Point is, we need to make sure that the legal requirements for paying people are legitimate.
    - Randy - we DO have an organization, and an organization bank account. What is missing is that the LocalDev Foundation doesn&#39;t have officers, governance, etc.
  - Truls - not sure every one of the sponsors and everyone supporting. We need to remember who is supporting us with the biggest packages or the largest financial support. Maybe they have a part in the organization and how it&#39;s structured. Not sure how it is with Platform.Sh, but they should get some input into the LocalDev Foundation stuff.
    - Randy - relationship of sponsorship levels with level of governance is a great concept to remember. I haven&#39;t even gotten to the 100+ sponsors on the website yet. But it&#39;s valid. How much you give vs. governance. This IS a small thing. It&#39;s not ever going to get &quot;big&quot; or maybe I&#39;m understating that, but small is easier than big.
  - Andrew Berry - re: how to frame this with the community. This is very much on the same path you should go. The first question is, &quot;is Platform.sh buying DDEV?&quot; so make sure we answer that. Lay out the top three roadmap in terms of what you&#39;ll do over the next 6 months. The concern is now Randy is committed only to the features Platform.sh wants, and not features the community wants.
  - Benny Mack - what Andrew said is important, esp. Re: the different technology stacks. Yes there will be Symfony/Platform integration. And yes that&#39;s important. It makes the DDEV brand stronger. Clarify that that doesn&#39;t mean that integrations are going away or are less important.
    - Randy - integrations are really open now; it&#39;s been well-received and people can change them. That&#39;s one of the important things here. Doesn&#39;t mean Platform may not come up with a different
  - Truls - very good to tell everyone using DDEV and the contributors, too - that they have the possibility of being part of the decisions and being part of the roadmap. Getting more contributors means you can be part of the change, too.
    - Randy - the way I&#39;ve dealt with the roadmap is that I work on what I perceive people are asking after the most. Node support and Python support are what Platform asked after, but for example Node support is something that the community has also clamored for.
- Followup from budget group meeting [https://docs.google.com/document/d/1YxU6N5m2qVfVJSSYXgjBbdPM6nBz9X4YjR0DWZIeMLw/edit?usp=sharing](https://docs.google.com/document/d/1YxU6N5m2qVfVJSSYXgjBbdPM6nBz9X4YjR0DWZIeMLw/edit?usp=sharing)
  - We&#39;re punting on this in favor of time, please check out on your own time.
- Review of v1.19 features (coming soon)
  - RC1 published yesterday, LOTS of new features.
  - One of the big ones is there are &quot;third-party&quot; services that are becoming more integrated. You can now use &quot;ddev get&quot; and &quot;ddev list&quot; to show these separate contrib repos… i.e. for Solr, Redis, etc. DDEV-Contrib++ - Contrib will be used for recipes that can&#39;t be standardized. These are for truly maintained things.
  - There is a call out for 1.19 testing. Please test if you have the ability. Issue is: [https://github.com/drud/ddev/issues/3645](https://github.com/drud/ddev/issues/3645)
  - We&#39;re a little ahead of our Mid-March schedule, but let&#39;s see. :)
- Questions:
  - When should PHP8.0 become the default PHP version?
    - Older versions need to be supported, and make it clear how it changes.
    - Randy&#39;s note - depending on different projects, it will work (ex.g. If you spin up a Laravel and it sets its default, Drupal 10 does its, etc.)
    - **DECISION: Let&#39;s do it in 1.20.**
  - Should any other defaults change in upcoming ddev v1.19.0?
    - NVM is installed in 1.19 - and it will be changeable to any _supported_ version. 16 is the latest LTS default. NVM only comes into play if you change it.
    - Truls - I was wondering about the PHP supported-versions. Should we follow that to make it the default? I think it&#39;s OK to change PHP now, even. (There is additional support for this in the chat; that is, people want to keep the cadence with PHP.)
    - Randy - we do have a way to support older versions, remember we&#39;re talking about the default. And we have ways to get really old supported versions as necessary. Remember Drupal 6 projects get a better/more sensible default. We are here to solve Developer problems, NOT PHP problems. We want a developer to be able to spin up an ancient project so they can upgrade it, maintain it, etc.
  - Highest priorities for v1.20?
    - Christian Münch - would like a new router replacement for ddev-router (perhaps with Traefik). [https://github.com/drud/ddev/issues/3458](https://github.com/drud/ddev/issues/3458) - this is a complex task, but we (Christian&#39;s org) has issues with the router. This could potentially break all the HTTP\_EXPOSEs that are out there.
    - Chris &amp; Andy Blum - you&#39;re the visionary here. You&#39;re coming up with stuff we didn&#39;t know we needed.
  - Guidance on test-runner replacement (Windows/macOS)
    - Punting for time.
  - Nominations for upstream projects to donate to with 10% of GitHub Sponsors
    - Already sponsor debsury for PHP
    - Already sponsor Xdebug
    - I checked with Colima/Lima - and they&#39;re sponsored by Rancher so they&#39;re OK.
    - Chris Wells: suggested NVM
      - https://github.com/sponsors/ljharb
    - Randy will open an issue for this.
- Randy ends with a final thank-you; see you in May!

# Recording

Recording is at https://www.dropbox.com/s/0f8rq2k05vk3zrs/20220302%20DDEV%20Advisory%20Group.mp4?dl=0 .

**Comment by rfay** [comment]
I added the video recording
